tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/Services/FailureAnalysis/CacheFailureClassifier.cs

```diff
@@ -0,0 +1,73 @@
+﻿using Microsoft.TeamFoundation.Build.WebApi;
+using Microsoft.VisualStudio.Services.WebApi;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Azure.Sdk.Tools.PipelineWitness.Services.FailureAnalysis
+{
+    public class CacheFailureClassifier : IFailureClassifier
+    {
+        private class FailureClassifier
+        {
+            public readonly string MessageContains;
+            public readonly string FailureName;
+
+            public FailureClassifier(string messageContains, string failureName)
+            {
+                this.MessageContains = messageContains;
+                this.FailureName = failureName;
+            }
+
+            public bool IsFailure(string message)
+            {
+                return message.StartsWith(MessageContains);
+            }
+        }
+
+        private static readonly FailureClassifier[] failureClassifiers = new FailureClassifier[]
+        {
+            new FailureClassifier("Chunks are not arriving in order or sizes are not matched up", "Cache Chunk Ordering" ),
+            new FailureClassifier("The task has timed out", "Cache Task Timeout" ),
+            new FailureClassifier("Service Unavailable", "Cache Service Unavailable"),
+            new FailureClassifier("The HTTP request timed out after", "Cache Service HTTP Timeout"),
+            new FailureClassifier("Access to the path", "Cache Cannot Access Path"),
```

Comment:
Is there a chance `Access to the path` might be too large a net for this specific type of failure?

Reply:
This is unlikely. The messages being checked are only those prefaced by `##[error]` and not all messages in the log.